src/main/java/uk/gov/hmcts/reform/sendletter/blob/component/BlobStitch.java

```diff
@@ -0,0 +1,110 @@
+import uk.gov.hmcts.reform.sendletter.services.pdf.PdfCreator;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Component
+public class BlobStitch {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BlobStitch.class);
+    private final PdfCreator pdfCreator;
+    private final SasTokenGeneratorService sasTokenGeneratorService;
+    private final BlobManager blobManager;
+    private static final String SEPARATOR = "_";
+    private static final String DEST_DIRECTORY = "/var/tmp/";
+
+    public BlobStitch(BlobManager blobManager, SasTokenGeneratorService sasTokenGeneratorService,
+                      PdfCreator pdfCreator) {
+        this.blobManager = blobManager;
+        this.sasTokenGeneratorService = sasTokenGeneratorService;
+        this.pdfCreator = pdfCreator;
+    }
+
+    public DeleteBlob stitchBlobs(PrintResponse printResponse) throws IOException {
+        List<Doc> docs = new ArrayList<>();
+        var deleteBlob = new DeleteBlob();
+        if (printResponse != null && printResponse.printJob != null && printResponse.printJob.documents != null
+            && printResponse.printUploadInfo != null && printResponse.printUploadInfo.uploadToContainer != null) {
+
+            var containerName = printResponse.printJob.containerName;
+            var serviceName = printResponse.printJob.service;
+            LOG.info("getPdfInfo serviceName {}, containerName {}", serviceName, containerName);
+
+            //generate sasToken
+            var sasToken = sasTokenGeneratorService.generateSasToken(serviceName);
+
+            //convert pdf to list of byte[]
+            List<String> blobs = new ArrayList<>();
+            for (var document : printResponse.printJob.documents) {
+                blobs.add(document.uploadToPath);
+                docs.add(getPdfDocuments(containerName, sasToken, document));
+            }
+            blobs.add(printResponse.printUploadInfo.manifestPath);
+            deleteBlob.setServiceName(serviceName);
+            deleteBlob.setContainerName(containerName);
+            deleteBlob.setBlobName(blobs);
+            //create final pdfname
+            var finalPdfName = generatePdfName(printResponse.printJob.type, serviceName,
+                    printResponse.printJob.id);
+
+            //pdf file with local file location
+            var finalPdfPath = DEST_DIRECTORY + finalPdfName;
+
+            // Create final pdf file from byte[] .e.g stitched
+            try (var fileOutputStream = new FileOutputStream(finalPdfPath)) {
+                fileOutputStream.write(pdfCreator.createFromBase64PdfWithCopies(docs));
+            }
+
+            // Create the container and return a container client object
+            var containerClient = blobManager.getContainerClient("processed");
```

Comment:
maybe pull this string into a static property?

Reply:
Done